Training/Training/FSM6A.c

```diff
@@ -0,0 +1,83 @@
+#define _CRT_SECURE_NO_WARNINGS  1
+#include <stdio.h>
+#include<string.h>
+/* Demo program to implement a Mealy machine.
+*  This machine detects a pattern 0110 or 1101 in an input stream.
+*/
+
+// Define states of the Mealy machine
+typedef enum {
+   S0,  // Initial state
+   S1,  // After '0'
+   S2,  // After '01'
+   S3,  // After '011'
+   A,   // After '110'
+   B,   // After '11'
+   C    // After '1'
+} State;
+
+// Function to get the next state and output based on the current state and input
+State nextMealyState (State currentState, int input, int* output) {
+   switch (currentState) {
+      case S0:
+         // Transition to S1 after '0'
+         // Transition to C after '1'
+         return input == 0 ? (*output = 0), S1 : ((*output = 0), C);

```

Comment:
output* = 0 is a statement that is common across all cases. You can take it outside the switch.